x-pack/plugins/security_solution/public/siem_migrations/rules/components/header_buttons/index.tsx

```diff
@@ -30,55 +30,51 @@ export interface HeaderButtonsProps {
   onMigrationIdChange: (selectedId?: string) => void;
 }
 
-const HeaderButtonsComponent: React.FC<HeaderButtonsProps> = ({
-  migrationsIds,
-  selectedMigrationId,
-  onMigrationIdChange,
-}) => {
-  const migrationOptions = useMemo(() => {
-    const options: Array<EuiComboBoxOptionOption<string>> = migrationsIds.map((id, index) => ({
-      value: id,
-      'data-test-subj': `migrationSelectionOption-${index}`,
-      label: i18n.SIEM_MIGRATIONS_OPTION_LABEL(index + 1),
-    }));
-    return options;
-  }, [migrationsIds]);
-  const selectedMigrationOption = useMemo<Array<EuiComboBoxOptionOption<string>>>(() => {
-    const index = migrationsIds.findIndex((id) => id === selectedMigrationId);
-    return index !== -1
-      ? [
-          {
-            value: selectedMigrationId,
-            'data-test-subj': `migrationSelectionOption-${index}`,
-            label: i18n.SIEM_MIGRATIONS_OPTION_LABEL(index + 1),
-          },
-        ]
-      : [];
-  }, [migrationsIds, selectedMigrationId]);
+export const HeaderButtons: React.FC<HeaderButtonsProps> = React.memo(
+  ({ migrationsIds, selectedMigrationId, onMigrationIdChange }) => {
+    const migrationOptions = useMemo(() => {
+      const options: Array<EuiComboBoxOptionOption<string>> = migrationsIds.map((id, index) => ({
+        value: id,
+        'data-test-subj': `migrationSelectionOption-${index}`,
+        label: i18n.SIEM_MIGRATIONS_OPTION_LABEL(index + 1),
+      }));
+      return options;
+    }, [migrationsIds]);
```

Comment:
Since the `migrationIds` received here are already filtered, the `index` does not correspond to the correct migration `number`.
Could this component receive the migration stats returned by the `useLatestStats` hook, instead of the ids alone? This hook returns the `RuleMigrationStats[]` type, which already has the `number` property.

Reply:
sounds good, done!